src/settings.cpp

```diff
@@ -1749,9 +1744,10 @@ void GetSettingSaveLoadByPrefix(const char *prefix, std::vector<SaveLoad> &savel
  * @return Pointer to the setting description of setting \a name if it can be found,
  *         \c nullptr indicates failure to obtain the description.
  */
-static const SettingDesc *GetCompanySettingFromName(const char *name)
+static const SettingDesc *GetCompanySettingFromName(std::string_view name)
 {
-	if (strncmp(name, "company.", 8) == 0) name += 8;
+	static const std::string_view company_prefix = "company.";
```

Comment:
Can a string be stored in a `string_view`? Where is the real string stored?

Shouldn't this just be a `std::string`?

Reply:
Yes.  string_view is like span but for string-like things, and in this case it is initialized with a C-string instead of converting the C-string to a std::string and then to a std::string_view. The advantage of using string_view here is that information of the length that is available at compile time does not get lost, compared to using a `const char *`, which makes later size checks/comparisons cheaper. And that calls to `StrStartsWith` do not need to (implicitly) convert data.

Reply:
Sometimes it seems you and I are not talking the same language ;) You do realise I reviewed enough of your PRs to know what `std::string_view` does right? Not what I was asking about ;)

What I am asking, is that `string_view` points (but does not contain) a string. So is it safe to assign a value to `string_view` like this, or do we need to store it in a proper variable first? For `std::span` I have been told it is UB, so is it a similar problem with `string_view`?

Reply:
To answer my own question: "it depends". https://en.cppreference.com/w/cpp/string/basic_string_view gives a nice example of good and bad usage, which can be best seen here: https://godbolt.org/z/c6YcG1Tzj . Annoyingly, GCC doesn't complain about this at all. Clang does, so that is something at least :)

So from my understanding, your usage is fine, as the c-string is in the `.data`, and as such remains valid. So the view around it remains valid too. If it would be anything that requires any runtime setup, it will not be valid.

Such a thin line to walk .. parts of C++ that still buzz my ears :)